Bottom: 3159dd6d8df231eeff5eec88274679b7afd5f91a
Top:    324c9032f2dbab2b41f23e50cfdd34c2fb0530c5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-02 14:06:19 +0000

Refresh of move-polygon-clipped-test-out-

---

diff --git a/src/draw.c b/src/draw.c
index 4a28844..a2e86a8 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -794,9 +794,6 @@ poly_callback (const BoxType * b, void *cl)
   struct poly_info *i = cl;
   PolygonType *polygon = (PolygonType *)b;
 
-  if (!polygon->Clipped)
-    return 0;
-
   set_layer_object_color (i->layer, (AnyObjectType *) polygon);
 
   gui->graphics->draw_pcb_polygon (Output.fgGC, polygon, i->drawn_area);
diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 9ea5094..17b1868 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -259,6 +259,9 @@ should_compute_no_holes (PolygonType *poly, const BoxType *clip_box)
 void
 common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip_box)
 {
+  if (polygon->Clipped == NULL)
+    return;
+
   if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
     gui->graphics->thindraw_pcb_polygon (gc, polygon, clip_box);
   else
@@ -279,6 +282,9 @@ common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip
 void
 common_fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 {
+  if (polygon->Clipped == NULL)
+    return;
+
   if (!poly->NoHolesValid)
     {
       /* If enough of the polygon is on-screen, compute the entire
